Marlin/src/feature/easythreed_ui.cpp

```diff
@@ -0,0 +1,242 @@
+  static PrintFlag print_key_flag = PF_START;
+
+  const millis_t ms = millis();
+
+  switch (key_status) {
+    case KS_IDLE:
+      if (print_flag && !printingIsActive()) {                      // Reset states if printing state went to inactive
+        print_flag = false;
+        print_key_flag = PF_START;
+        blink_interval_ms = LED_ON;
+      }
+      if (!READ(BTN_PRINT)) {                                       // Print/Pause/Resume button pressed?
+        key_time = ms;                                              // Save start time
+        key_status++;                                               // Go to debounce test
+      }
+      break;
+
+    case KS_PRESS:
+      if (ELAPSED(ms, key_time + BTN_DEBOUNCE_MS))                  // Wait for debounce interval to expire
+        key_status = READ(BTN_PRINT) ? KS_IDLE : KS_PROCEED;        // Proceed if still pressed
+      break;
+
+    case KS_PROCEED:
+      if (!READ(BTN_PRINT)) break;                                  // Wait for the button to be released
+      key_status = KS_IDLE;                                         // Ready for the next press
+      if (PENDING(ms, key_time + 1200 - BTN_DEBOUNCE_MS)) {         // Register a press < 1.2 seconds
+        switch (print_key_flag) {
+          case PF_START:                                            // The "Print" button starts an SD card print
+            if (!printingIsActive()) {                              // Not already printing?
+              card.mount();                                         // Force SD card to mount - now!
+              if (!card.isMounted) {                                // Failed to mount?
+                blink_interval_ms = LED_OFF;                        // Turn off LED
+                print_flag = false;                                 // Clear previous print flag
+                return;                                             // Bail out
+              }
+              card.ls();                                            // List all files to serial output
+              const uint16_t filecnt = card.countFilesInWorkDir();  // Count printable files in cwd
+              if (filecnt == 0) return;                             // None are printable?
+              card.selectFileByIndex(filecnt);                      // Select the last file according to current sort options
+              card.openAndPrintFile(card.filename);                 // Start printing it
+              blink_interval_ms = LED_BLINK_2;                      // Blink the indicator LED at 1 second intervals
+              print_flag = true;                                    // Tell EasythreedUI printing should be active
+              print_key_flag = PF_PAUSE;                            // The "Print" button now pauses the print
+            }
+            break;
+          case PF_PAUSE:                                            // Pause printing (not currently firing)
```

Comment:
If this is not firing, look for where `print_key_flag` is being changed incorrectly.